client/src/components/Applications/AppForm/AppForm.js

```diff
@@ -57,6 +57,11 @@ const AppDetail = () => {
     shallowEqual
   );
 
+  const categoryDict = useSelector(state => state.apps.categoryDict);
+  const hasApps = Object.keys(categoryDict).some(
+    category => categoryDict[category].length > 0
```

Comment:
Same thing here, will this have an issue if the category does not exist in the dictionary or if the value of this key is null/undefined? (not sure if that's a possibility without other major things breaking)